docs/adr/2023-07-05_024-offline-mode.md

```diff
@@ -0,0 +1,38 @@
+---
+slug: 25
+title: | 
+  1. Offline mode
+authors: []
+tags: []
+---
+## Status
+N/A
+
+## Context
+
+Currently, the Hydra CLI requires a Layer 1 Cardano node running in order to operate; The L1 node is needed to submit and watch for L1 transactions. Generally speaking, the transactions watched are for learning the state of peer Hydra nodes. The transactions submitted are to transition between states (e.g. after submitting a Commit tx to the L1, a node watches to see when all other nodes have also Comitted.)
+
+There are applications for the Hydra node where interaction with a Layer 1 chain is unnecessary. Sundaeswap is developing the Gummiworm protocol, a Layer 2 rollup chain for Cardano. Gummiworm intends to have dedicated "validator node" entities, that validate & challenge the full history of transactions for a given Layer 2 Gummiworm session. These validator nodes don't need to connect to any peers, and each will be driven by a single client. SundaeSwap intends to use modified Hydra nodes, running in offline mode, to act as these validator nodes. SundaeSwap intends to upstream the offline mode patches to mainline Hydra.
+
+An optional offline mode would remove the dependency on the L1 Cardano node, for applications like Gummiworm where it is unneeded. This would be useful for other Layer 2s that build on top of Hydra instead of duplicating its efforts, and for anyone who wants to easily validate a set of Cardano transactions.
+
+## Decision
+We would like to introduce an offline mode command to the Hydra CLI, as an alternative to the default online mode. Hydra node, in online mode, depended on querying a Cardano node for Era History information and Genesis paramters. In offline mode, this information will instead be supplied via two sub-arguments to the offline-mode parameter, which each specify the path to a JSON file.
+
+Commit actions will not fail due to insufficient resources on the L1 chain, and may be used to specify the initial L2 UTXO state.
```

Comment:
Commits are indeed used to define the initial state of the L2 ledger, and can be empty. What's missing in this ADR is perhaps some diagram showing how the Hydra on-chain protocol is "collapsed" to a simpler version, so that the hydra-node and `HeadLogic` can be run as-is without requiring any modification. Alternatively, this off-chain mode could define its own logic without any on-chain component, reusing the underlying `IsTx` and `Ledger` stuff?